src/components/SetGuestUsername.vue

```diff
@@ -6,27 +6,32 @@
 <template>
 	<div class="username-form">
 		<!-- eslint-disable-next-line vue/no-v-html -->
-		<h3 v-html="displayNameLabel" />
+		<h3 v-if="!compact" v-html="displayNameLabel" />
 
-		<NcButton v-if="!isEditingUsername"
+		<NcButton v-if="!isEditingUsername && !compact"
 			@click="toggleEdit">
 			{{ t('spreed', 'Edit display name') }}
 			<template #icon>
-				<Pencil :size="20" />
+				<IconPencilOutline :size="20" />
 			</template>
 		</NcButton>
 
-		<NcTextField v-else
-			ref="usernameInput"
-			v-model="guestUserName"
-			:placeholder="t('spreed', 'Guest')"
-			class="username-form__input"
-			:show-trailing-button="!!guestUserName"
-			trailing-button-icon="arrowEnd"
-			:trailing-button-label="t('spreed', 'Save name')"
-			@trailing-button-click="updateDisplayName"
-			@keydown.enter="updateDisplayName"
-			@keydown.esc="toggleEdit" />
+		<div v-else class="username-form__display-name">
+			<IconAccountOutline class="username-form__display-name-icon" :size="20" />
+			<NcTextField
+				ref="usernameInput"
+				v-model="guestUserName"
+				:placeholder="t('spreed', 'Guest')"
+				class="username-form__input"
+				:label-outside="!compact"
+				:label="compact ? t('spreed', 'Display name (required)') : undefined"
```

Comment:
Maybe always show it for consistency? I don't mind having it as a guest in Rightsidebar